src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/services/CredentialDefinitionServiceTest.java

```diff
@@ -0,0 +1,167 @@
+package didentity.amos.digitalIdentity.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+@DataJpaTest
+public class CredentialDefinitionServiceTest {
+
+    private CredentialDefinitionService credentialDefinitionService;
+
+    @Mock
+    private LissiApiService lissiApiServiceMock;
+
+    @Mock
+    private ResourceService resourceServiceMock;
+
+    private AutoCloseable autoCloseable;
+
+    @BeforeEach
+    void setUp() {
+        autoCloseable = MockitoAnnotations.openMocks(this);
+
+        credentialDefinitionService = new CredentialDefinitionService();
+        credentialDefinitionService.setLissiApiService(lissiApiServiceMock);
+        credentialDefinitionService.setResourceService(resourceServiceMock);
+    }
+
+    @Test
+    void testCreate_withValidResponseOfLissiApiService_shouldCreateCredentialAndReturn201() {
+        // Arrange
+        this.intallLissiApiServiceMock_createCredentialDefinition();
+        this.intallRessourceServiceMock_getDummyPng();
+
+        String alias = "test";
+        String comment = "test";
+        String schemaId = "test";
+
+        String expected = "Succesfully created new credential definition.";
+        HttpStatus httpStatusExpected = HttpStatus.CREATED;
+
+        // Act
+        ResponseEntity<String> response = credentialDefinitionService.create(alias, comment, schemaId);
+        String responseAsString = response.getBody();
+        HttpStatus httpStatus = response.getStatusCode();
+
+        // Assert
+        assertEquals(expected, responseAsString);
+        assertEquals(httpStatusExpected, httpStatus);
+    }
+
+    @Test
+    void testCreate_withWrongParametersForCreateCredentialDefinition_shouldNotCreateCredentialAndReturn500() {
+        // Arrange
+        this.intallRessourceServiceMock_getDummyPng();
```

Comment:
```suggestion
        this.intallRessourceServiceMock_getDummyPng();
        // make it fail
        Mockito.when(lissiApiServiceMock.createCredentialDefinition(anyString(), anyString(), anyString(), anyString(),
                any())).thenReturn(null);
```

Reply:
I will add this as an extra test. Very good suggestion!

Reply:
see above. you do not need to add it as an extra test